pyproject.toml

```diff
@@ -41,3 +41,20 @@ build-backend = "setuptools.build_meta:__legacy__"
 [tool.setuptools_scm]
 version_scheme = "guess-next-dev"
 local_scheme = "no-local-version"
+
+[tool.coverage.run]
+omit = [
+    "scheduler/alembic/*"
+]
+
+[tool.coverage.report]
+exclude_lines = [
+  "pragma: no cover",
+  "def __repr__",
+  "def __str__",
+  "raise AssertionError",
+  "raise NotImplementedError",
+  "if __name__ == .__main__.:",
+  "if TYPE_CHECKING:",
+  "if typing.TYPE_CHECKING:"
+]
```

Comment:
I don't see a reason for excluding most of these. `__repr__` and `__str__` should have coverage because it's just code that can have bugs and should be tested. The same for `__name__ == .__main__.`. I can't think of a good usecase for `raise AssertError` but if it is used there should be a test case for it.

Excluding `if TYPE_CHECKING` is actually valid (although not used in the code). The `raise NotImplementedErrror` is also valid, although my opinion is that interface classes aren't useful in Python most of the time, but that's a discussion for another time...